### PR TITLE
KHWE-76: Integrate login page with firebase

### PR DESCRIPTION
**Pull request by lubovgribiniyk**
1. Implement Auth Service
2. Change Router to add support of protected routes
3. Create separated layouts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,16 +7,17 @@
     "@material-ui/core": "3.5.1",
     "@material-ui/icons": "3.0.1",
     "@material-ui/lab": "3.0.0-alpha.23",
+    "firebase": "^5.5.9",
```
Comment (pavelbaranchuk) [inline package.json]: Let's fix the package version and remove ^.
Same for react-redux. What do you think?

File: src/store/reducers/auth.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        user: action.user,
+        isLoaded: true
+      };
+    case USER_NOT_FOUND:
+      return {
+        ...state,
+        isLoggedIn: false,
+        isLoaded: true
+      };
+    case "SIGN_OUT":
```
Comment (ihor4568) [inline src/store/reducers/auth.js]: please use a constant

Comment (an2323) [reply src/store/reducers/auth.js]: fixed

Comment (lubovgribiniyk) [reply src/store/reducers/auth.js]: fixed

File: src/components/routes/PublicRoute.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+import React from "react";
+import { Route } from "react-router-dom";
+import PublicLayout from "../shared/PublicLayout";
+
+function PrivateRoute({ component: Component, ...rest }) {
```
Comment (ihor4568) [inline src/components/routes/PublicRoute.js]: maybe PublicRoute ?

File: src/components/routes/PrivateRoute.js [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    <Route
+      {...rest}
+      render={props => {
+        if (!isLoaded) {
+          return null;
+        }
+
+        if (isLoaded && isLoggedIn) {
+          return (
+            <>
```
Comment (ihor4568) [inline src/components/routes/PrivateRoute.js]: no need to wrap this inside fragment - you are returning only one element

File: src/components/routes/PublicRoute.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  return (
+    <Route
+      {...rest}
+      render={props => (
+        <>
```
Comment (ihor4568) [inline src/components/routes/PublicRoute.js]: no need to wrap this inside fragment - you are returning only one element

File: src/components/shared/AppBar.js [hunk 6]
```diff
@@ -31,6 +31,10 @@ const styles = theme => ({
       duration: theme.transitions.duration.enteringScreen
     })
   },
+  appBarDisabled: {
+    backgroundColor: "#ccc",
```
Comment (ihor4568) [inline src/components/shared/AppBar.js]: could we move this color to theme 
or use one from theme that already exists?

**Pull request merged**